src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,108 @@
+import { FC, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { PersonLink } from '../PersonLink';
+import cn from 'classnames';
+
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getHuman } from '../../utils/getHuman';
+
+export const PeoplePage: FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const { slug } = useParams();
+  const selectedPerson = slug;
+  const isPeopleListEmpty = people.length === 0 && !loading && !error;
+
+  const renderHuman = (humanName: string | null) => {
+    const human = getHuman(humanName, people);
+
+    if (!human) {
+      return '-';
+    }
+
+    if (typeof human !== 'string') {
+      return <PersonLink person={human} />;
+    }
+
+    return humanName;
+  };
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setError(true))
```

Comment:
Consider resetting the `people` state to an empty array when an error occurs. This will ensure that any previously loaded data is cleared if a new error happens during a subsequent fetch attempt.